seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/SchemaUtil.java

```diff
@@ -69,9 +71,14 @@ public static Column toSeaTunnelType(BasicTypeDefine<DataType> typeDefine) {
     }
 
     public static DataField getDataField(List<DataField> fields, String fieldName) {
-        return fields.parallelStream()
-                .filter(field -> field.name().equals(fieldName))
-                .findFirst()
-                .get();
+        Optional<DataField> firstField =
+                fields.parallelStream().filter(field -> field.name().equals(fieldName)).findFirst();
+        if (firstField.isPresent()) {
+            return firstField.get();
+        }
+        throw new SeaTunnelException(
+                String.format(
+                        "Con not get the DataField named: [%s] in sink schema. The schema of paimon is case-sensitive in default. Please check it.",
+                        fieldName));
```

Comment:
Why not just list `fields` values in exception message? `case not match` just only one situation which can cause this exception.
You can reuse this error:
https://github.com/apache/seatunnel/blob/c46e16a128fa7ac5a4f2c507296a882279182a10/seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonError.java#L251

Reply:
OK